src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets

```diff
@@ -21,23 +19,25 @@
       XmlInputPaths="@(AsnXml)"
       OutputPaths="@(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs')" />
 
-    <Exec Condition="'$(TargetOS)'=='windows'"
+    <Exec Condition="$([MSBuild]::IsOsPlatform('windows')) == 'true'"
       IgnoreExitCode="true"
       StandardOutputImportance="Low"
       Command="$(SystemRoot)\System32\fc.exe /a @(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs') @(AsnXml -> '%(Identity).cs')">
       <Output TaskParameter="ExitCode" ItemName="_AsnXmlDiffCode" />
     </Exec>
 
-    <!-- TODO: Call diff on Unix -->
-    <ItemGroup Condition="'$(TargetOS)'!='windows'">
-      <_AsnXmlDiffCode Include="1" />
-    </ItemGroup>
+    <Exec Condition="$([MSBuild]::IsOsPlatform('windows')) != 'true'"
+      IgnoreExitCode="true"
+      StandardOutputImportance="Low"
+      Command="/usr/bin/diff -q -a @(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs') @(AsnXml -> '%(Identity).cs')">
+      <Output TaskParameter="ExitCode" ItemName="_AsnXmlDiffCode" />
+    </Exec>
 
     <Copy
       Condition="'@(_AsnXmlDiffCode)' != '0'"
-      SourceFiles="@(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs')" 
+      SourceFiles="@(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs')"
       DestinationFiles="@(AsnXml -> '%(Identity).cs')" />
-    
-    <Warning Condition="'@(_AsnXmlDiffCode)' != '0'" Text="AsnXml regenerated files, be sure to check them in: @(AsnXml -> '%(Identity).cs')" />
+
+    <Message Condition="'@(_AsnXmlDiffCode)' != '0'" Text="AsnXml regenerated files, be sure to check them in: @(AsnXml -> '%(Identity).cs')" Importance="high" />
```

Comment:
Since this now runs as part of the normal build (`./build.sh` from the root) and we treat warnings as errors there, the warning would fail the build if the files were regenerated. The message stops that from happening, but unfortunately is not quite as prominent and easy to miss.

Reply:
Warning (as error) might be better, since it means that the source tree and the build are not in a consistent state.  If you've changed one of these files it means you have to up-enter.  If you forget to add one to a commit you fail the build in the PR.

Reply:
> Warning (as error) might be better

Hm, it seems strange to fail the build just to remind you to check in a file. Seems like friction for someone less familiar with how it works.

> If you've changed one of these files it means you have to up-enter.

I don't follow. You need to run the build twice if you change one of these files?

> If you forget to add one to a commit you fail the build in the PR.

Well, that's true of normal code changes as well assuming a test is written to cover the changes.

Reply:
Hmm. I thought on this a bit more and it makes some sense now if you think about it the way the reference source projects work. Ideally if the XML and .cs file are out-of-sync you want to fail in CI (which is what I think you were saying) much like forgetting to update / check in the ref source.

Reply:
In that case I would want to go full `<Error />` instead of `<Warning />` so that the error consistently happens whether you use `./build.sh` or `dotnet build` in a project directory (the latter is what I tend to use).

Reply:
Ah, but that's the beauty of Warning with Warn-As-Error.  When you're developing (and use `dotnet build` locally) it's not an error, just a "hey, pay attention" blob of yellow.  If you don't listen and try to check in then it's an error.

So I think Warning (with Warn-As-Error from CI) is still the good place to be.

If there are complaints we can adjust it to message normally and Error when some variable tells us it's an official build.  (Then you could still use `build.{cmd|sh} -allConfigurations -subset libs` for "I changed the xslt, regenerate everything" but `build.{cmd|sh} -ci` would produce an error in that situation.)  What I didn't like about that is that switching on the officialness of the build is a little fragile (if the variable changes and this file is missed, it just never sees an official build).  So maybe the increment would be it's a message for debug builds and a warning or error for release builds.

Reply:
Okay, I went back to a warning for now.